syntax = "proto3";

package inworld.ndkData;

message ClientOptions {
    string auth_url = 1;
    string load_scene_url = 2;
    string scene_name = 3;
    string api_key = 4;
    string api_secret = 5;
    string player_name = 6;
    string token = 7;
    string session_id = 8;
}

message AgentInfo {
    string BrainName = 1;
    string AgentId = 2;
    string GivenName = 3;
}

message SessionInfo {
    string SessionId = 1;
    string Token = 2;
    int64 ExpirationTime = 3;
    bool IsValid = 4;
}

message AgentInfoArray {
    repeated AgentInfo agent_info_list = 1;
}

enum ConnectionState {
    Idle = 0;
    Connecting = 1;
    Connected = 2;
    Failed = 3;
    Paused = 4;
    Disconnected = 5;
    Reconnecting = 6;
}
